challenge/dns01/nameserver.go

```diff
@@ -26,6 +26,10 @@ var defaultNameservers = []string{
 // recursiveNameservers are used to pre-check DNS propagation.
 var recursiveNameservers = getNameservers(defaultResolvConf, defaultNameservers)
 
+// networkStack is used to define which IP stack will be used. The default is
+// both IPv4 and IPv6. Set to "ipv4" for IPv4 only, and "ipv6" for IPv6 only.
+var networkStack = "both"
```

Comment:
I believe an enum would be better here:

```suggestion
type networkStack int

const (
	DefaultNetworkStack networkStack = iota
	IPv4Only
	IPv6Only
)

// currentNetworkStack is used to define which IP stack will be used. The default is
// both IPv4 and IPv6. Set to IPv4Only or IPv6Only to select either version.
var currentNetworkStack = DefaultNetworkStack
```

Reply:
I'm not a fan of a Go "enum": it's not a real enum, it's just a weak list of values.

Reply:
@ldez what do you want to see?

Reply:
As this PR currently only updates the `dns01` package (and leaves `http01` and `tlsalpn01` as a TODO), maybe we can find a way to pass the stack selection to `dns01.NewChallenge`, `http01.NewChallenge` and `tlsalpn01.NewChallenge` as an option?

Looking at `dns01.NewChallenge`, it already accepts a vararg for options:

https://github.com/go-acme/lego/blob/1cad41db658621daca9c0950c731a66308ae5e4b/challenge/resolver/solver_manager.go#L51-L54

Maybe the `networkStack` can move into the `cmd` package (in whatever form), and passing the selection as option to the challenge solver can be done (centrally) in `setupChallenges`?

https://github.com/go-acme/lego/blob/1cad41db658621daca9c0950c731a66308ae5e4b/cmd/setup_challenges.go#L25-L26

Reply:
> (and leaves `http01` and `tlsalpn01` as a TODO)

It does not leave them as todo items. This PR is in draft because I wanted guidance prior to actually working on the full solution. At this point I'm getting conflicting feedback and would like a clear direction. I need this feature and am willing to put in the work, but I don't want to flail around to get it done.

Reply:
Sorry for the miscommunication, English isn't my first language. I didn't mean "left as TODO for others", but "for this to work, we need to think of http01 and tlsalpn01 as well".